File: CMOWebApi.Models/AdminModel/MasterModel/ChildPageMasterModel.cs
using CMOWebApi.Models.GeneralModel;
using System;

namespace CMOWebApi.Models.AdminModel.MasterModel
{
    public class ChildPageMasterModel
    {
        public long Id { get; set; }
        public Nullable<long> PageCode { get; set; }
        public Nullable<long> PageDescriptionCategory { get; set; }
        public string Description { get; set; }
        public string PDFURL { get; set; }
        public string ImageURL { get; set; }
        public bool IsActive { get; set; } = true;
        public bool IsDeleted { get; set; } = false;
        public Nullable<int> PageTypeCode { get; set; }
        public Nullable<int> LookupTypeId { get; set; }
        public Nullable<long> SchemeTypeCode { get; set; }
        public string URL { get; set; }
        public int ManualTypeCode { get; set; }
    }

    public class ChildPageMasterViewModel
    {
        public long Id { get; set; }
        public Nullable<int> Code { get; set; }
        public Nullable<long> PageCode { get; set; }
        public Nullable<long> PageDescriptionCategory { get; set; }
        public string Description { get; set; }
        public string PDFURL { get; set; }
        public string ImageURL { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<int> ModifiedBy { get; set; }
        public System.DateTime CreatedDate { get; set; }
        public Nullable<System.DateTime> ModifiedDate { get; set; }
        public bool IsActive { get; set; }
        public bool IsDeleted { get; set; }
        public string PageTitle { get; set; }
        public string ApplicationTitle { get; set; }
        public string AplicationDisplayTitle { get; set; }
        public string PageTypeName { get; set; }
        public string MenuTitle { get; set; }
        public string DefaultUrl { get; set; }
        public Nullable<int> PageTypeCode { get; set; }
        public Nullable<int> LookupTypeId { get; set; }
        public Nullable<long> SchemeTypeCode { get; set; }
        public string PageDescriptionCategoryName { get; set; }
        public string URL { get; set; }
        public int ManualTypeCode { get; set; }
        public string ManualType { get; set; }
    }

    public class PageManualModel
    {
        public Nullable<int> LookupTypeId { get; set; }
        public Nullable<long> SchemeTypeCode { get; set; }
        public string menuName { get; set; }
    }

    public class ChildPageFilterModel:IndexModel
    {
        public long PageCode { get; set; }
        public int PageTypeCode { get; set; }
        public string ApplicationCode { get; set; }
        public int ManualTypeCode { get; set; }
    }

    public class PageMasterDetailModel
    {
        public long PageCode { get; set; }
        public long PageId { get; set; }
        public string PageTitle { get; set; }
        public string ApplicationTitle { get; set; }
        public string AplicationDisplayTitle { get; set; }
        public string PageTypeName { get; set; }
        public string MenuTitle { get; set; }
        public string DefaultUrl { get; set; }
    }

}

File: ANGULAR/ANGULAR/src/app/Shared/Model/Master/child-page-master.model.ts
import { IndexModel } from './../general-model';
export class ChildPageMasterModel {
  Id: number;
  PageCode: number | string;
  PageTypeCode: number | string;
  PageDescriptionCategory: number | string;
  Description: string;
  PDFURL: string;
  ImageURL: string;
  LookupTypeId: number | string;
  SchemeTypeCode: number | string;
  URL: string;
  ManualTypeCode: number | string;

  constructor() {
    this.Description = "";
  }
}

export class ChildPageMasterViewModel {
  Id: number;
  Description: string;
  PageTitle: string;
  ApplicationTitle: string;
  PageTypeName: string;
  MenuTitle: string;
  IsActive: boolean;
  PageDescriptionCategoryName: string;
  PDFURL: string;
  ImageURL: string;
  URL: string;
  ManualType: string;
}

export class PageMasterDetailModel {
  PageTitle: string;
  ApplicationTitle: string;
  PageTypeName: string;
  MenuTitle: string;
}

export class ChildPageFilterModel extends IndexModel {
  PageCode: number;
  PageTypeCode: number;
  ApplicationCode: string;
  ManualTypeCode: number | string;

  constructor() {
    super();
    this.ManualTypeCode = '0';
  }
}

export class PageManualModel {
  LookupTypeId: number;
  SchemeTypeCode: number;
  menuName: string;
  constructor(menuName = "", lookupTypeId = 0, schemeTypeCode = 0) {
    this.LookupTypeId = lookupTypeId;
    this.SchemeTypeCode = schemeTypeCode;
    this.menuName = menuName;
  }
}
export class AchievementSearchModel {
  AdmDepartmentCode: number;
  DepartmentCode: number;
  CategoryCode: number;
  SubCategoryCode: number;
  FromDate: string;
  ToDate: string;
  SearchKeyword: string;
  IsUrlRequired: boolean;
  IsImageRequired: boolean;
  IsPdfRequired: boolean;
  IsActive: boolean | null;
  IsVisibleInHome: boolean | null;
  IsVisibleInFront: boolean | null;
  IndexModel: IndexModel;
  CommonCategoryCode: number;
}
